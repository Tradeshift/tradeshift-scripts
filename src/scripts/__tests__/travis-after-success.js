import cases from 'jest-in-case';

// this removes the quotes around strings...
expect.addSnapshotSerializer({
	print: val => val,
	test: val => typeof val === 'string'
});

cases(
	'travis-after-success',
	({ version = '0.0.0-semantically-released', hasCoverageDir = true, branch = 'master' }) => {
		// beforeEach
		const { sync: crossSpawnSyncMock } = require('cross-spawn');
		const utils = require('../../utils');
		utils.resolveBin = (modName, { executable = modName } = {}) => executable;
		const originalExit = process.exit;
		process.exit = jest.fn();

		// tests
		crossSpawnSyncMock.mockClear();
		crossSpawnSyncMock.mockReturnValue({
			stdout: {
				toString() {
					return branch;
				}
			}
		});
		if (version) {
			utils.pkg.version = version;
		}
		utils.hasFile = filename => filename === 'coverage' && hasCoverageDir;
		require('../travis-after-success');
		expect(crossSpawnSyncMock).toHaveBeenCalledTimes(2);
		const [, secondCall] = crossSpawnSyncMock.mock.calls;
		const [script, calledArgs] = secondCall;
		expect([script, ...calledArgs].join(' ')).toMatchSnapshot();

		// afterEach
		process.exit = originalExit;
		jest.resetModules();
	},
	{
		'calls concurrently with both scripts': {},
		'does not do the autorelease script when the version is different': {
			version: '1.2.3'
		},
		'does not do the codecov script when there is no coverage directory': {
			hasCoverageDir: false
		},
		'adds dry-run flag when not running on master': {
			branch: 'dev-branch'
		}
	}
);
