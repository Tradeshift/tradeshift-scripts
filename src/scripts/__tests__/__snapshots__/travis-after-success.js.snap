// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`travis-after-success adds dry-run flag when not running on master 1`] = `concurrently --prefix [{name}] --names codecov,release --prefix-colors bgBlue.bold.reset,bgGreen.bold.reset "echo installing codecov && npx -p codecov -c 'echo running codecov && codecov'" "echo installing semantic-release && npx -p semantic-release@15 -c 'echo running semantic-release dry run && semantic-release --no-ci --dry-run'"`;

exports[`travis-after-success calls concurrently with both scripts 1`] = `concurrently --prefix [{name}] --names codecov,release --prefix-colors bgBlue.bold.reset,bgGreen.bold.reset "echo installing codecov && npx -p codecov -c 'echo running codecov && codecov'" "echo installing semantic-release && npx -p semantic-release@15 -c 'echo running semantic-release  && semantic-release --no-ci'"`;

exports[`travis-after-success does not do the autorelease script when the version is different 1`] = `concurrently --prefix [{name}] --names codecov --prefix-colors bgBlue.bold.reset "echo installing codecov && npx -p codecov -c 'echo running codecov && codecov'"`;

exports[`travis-after-success does not do the codecov script when there is no coverage directory 1`] = `concurrently --prefix [{name}] --names release --prefix-colors bgBlue.bold.reset "echo installing semantic-release && npx -p semantic-release@15 -c 'echo running semantic-release  && semantic-release --no-ci'"`;
